forge/db/models/Team.js

```diff
@@ -248,21 +248,112 @@ module.exports = {
                     // All Team owners
                     return this.memberCount(Roles.Owner)
                 },
-                projectCount: async function (projectTypeId) {
+                instanceCount: async function (projectTypeId) {
                     const where = { TeamId: this.id }
                     if (projectTypeId) {
                         if (typeof projectTypeId === 'string') {
                             projectTypeId = M.ProjectType.decodeHashid(projectTypeId)
+                        } else if (projectTypeId.id) {
+                            projectTypeId = projectTypeId.id
                         }
                         where.ProjectTypeId = projectTypeId
                     }
                     return await M.Project.count({ where })
                 },
+                instanceCountByType: async function (where = {}) {
+                    where = { ...where, TeamId: this.id }
+                    const counts = await M.Project.count({
+                        where,
+                        attributes: ['ProjectTypeId'],
+                        group: 'ProjectTypeId'
+                    })
+                    const result = {}
+                    for (const instanceType of Object.values(counts)) {
+                        result[M.ProjectType.encodeHashid(instanceType.ProjectTypeId)] = instanceType.count
+                    }
+                    return result
+                },
                 pendingInviteCount: async function () {
                     return await M.Invitation.count({ where: { teamId: this.id } })
                 },
                 deviceCount: async function () {
                     return await M.Device.count({ where: { TeamId: this.id } })
+                },
+                /**
+                 * Many functions require this.TeamType to exist and be fully populated.
+                 * Depending on the route taken, it is possible this property has not
+                 * been fully loaded. This does the work to ensure it is there if needed.
+                 */
+                ensureTeamTypeExists: async function () {
+                    if (!this.TeamTypeId) {
+                        await this.reload({ include: [{ model: M.TeamType }] })
```

Comment:
There were multiple places where we load a model and include Team, but limit it to the minimum of team attributes that are needed in that context. That would mean it didn't have `TeamTypeId` because it wasn't one of the selected attributes.

The reload ensures the full Team object is retrieved and the TeamType can be included.

I did go through a lot of the relevant models to ensure they included `TypeTypeId`, but wasn't 100% confident there wasn't still a `reload` lurking somewhere that would get missed. This is a safety net just in case.